vortex-array/src/array/sparse/mod.rs

```diff
@@ -41,18 +41,11 @@ impl SparseArray {
         values: Array,
         len: usize,
         indices_offset: usize,
-        fill_value: Scalar,
+        fill_value: ScalarValue,
     ) -> VortexResult<Self> {
         if !matches!(indices.dtype(), &DType::IDX) {
             vortex_bail!("Cannot use {} as indices", indices.dtype());
         }
-        if values.dtype() != fill_value.dtype() {
```

Comment:
I think it would be nice to keep a check like this but would have to add method to dtype to check if it is a valid scalar value for the type

Reply:
https://github.com/spiraldb/vortex/pull/958/files#diff-615981632b741f7a38cbfe010f9bdcb04b60a5692d139d53b613d36c74a93aebR49